### ISSUE TITLE
Transaction error Internal JSON-RPC error on Token view Send when not enough MATIC to cover gas fee

### ISSUE THREAD

**Issue by chrisleewilcox** [MetaMask/metamask-mobile#4801]
```
Given I am on Wallet view
And I am on Polygon Mainnet
And I have 0.0001 MATIC balance
And I have 2.01 USDT balance
When I tap on USDT in Tokens list
And I tap Send in Token view
And I tap on Account 2 (either in Recents or Transfer between my accounts)
And I tap Next
And I tap Use Max
And I tap Next
Then Confirm screen is displayed
And error/warning message indicating I need more MATIC to complete transaction (FAILED: no message, see CW4)
And Send is disabled (FAILED: Send is enabled, see CW4)
```
Tapping Send in this scenario will display the JSON-RPC error.

https://recordit.co/uz4NxOH2Pk

**Comment by blackdevelopa** [comment]
Hi Chris, I am unable to reproduce this anymore.
https://images.zenhubusercontent.com/141427485/e7cd6e0f-1527-4acc-b668-414f7c447180/polygon_bug_4801.mov
[Uploading... Screen Recording 2022-09-07 at 09.16.37.mov]

However, I think I have seen this error and will update in a bit

**Comment by blackdevelopa** [comment]
As shared in our Slack chat, I don't think this is the reason. This error seem elusive and pretty random. However, you are sure to see this when you try to speed up confirmed transactions.

**Comment by bschorchit** [comment]
I was able to reproduce it too on v5.6. 
I did not get get the warning message saying I don't have enough funds for the transaction (even though the gas estimated was higher then my Matic balance) and the send button was enabled. Clicking on the send lead to Transaction error Internal JSON-RPC error message. 

![IMG_8806.jpg](https://images.zenhubusercontent.com/61e705ae78964374054c3ed0/e74e4a45-ceb1-4006-8ad4-c5bad1b09094)
![IMG_8807.jpg](https://images.zenhubusercontent.com/61e705ae78964374054c3ed0/5c6c12b0-1a63-41e2-a601-52f7ffb795e3)
